### PR TITLE
RC5.0: Make using API trimming optional

### PR DESCRIPTION
**Pull request by scottn12**
Makes API trimming optional in RC5

Cherry pick of https://github.com/microsoft/FluidFramework/commit/23c2373b25dafc5952203352abb58eb1ed175de6

### CODE REVIEW

File: packages/framework/fluid-framework/api-report/fluid-framework.public.api.md [hunk 1]
```diff
@@ -26,9 +31,17 @@ import { PropertiesManager } from '@fluidframework/merge-tree/internal';
 import { PropertySet } from '@fluidframework/merge-tree/internal';
 import { ReferencePosition } from '@fluidframework/merge-tree/internal';
 import { ReferenceType } from '@fluidframework/merge-tree/internal';
+import { SchemaLibrary as SchemaLibrary_2 } from '../feature-libraries/schemaBuilderBase.js';
```
Comment (Josmithr) [inline packages/framework/fluid-framework/api-report/fluid-framework.public.api.md:34]: Have we checked the rollups generated for this package? This import statement is invalid. This is a known API-Extractor bug (https://github.com/microsoft/rushstack/issues/4507), but also often appears as an indicator of other things going wrong.

File: azure/packages/azure-service-utils/package.json [hunk 2]
```diff
@@ -14,12 +14,18 @@
 	"type": "module",
 	"exports": {
 		".": {
+			"trimmedAPI": {
```
Comment (tylerbutler) [inline azure/packages/azure-service-utils/package.json:17]: Can you add the comment field too?

Comment (scottn12) [reply azure/packages/azure-service-utils/package.json:17]: Good catch, fixed in latest

File: packages/service-clients/odsp-client/package.json
Comment (jason-ha) [inline packages/service-clients/odsp-client/package.json:1]: We should revert all of the osdp-client changes. None of these are needed. (It was never trimmed!)

File: packages/tools/devtools/devtools-core/package.json
Comment (jason-ha) [inline packages/tools/devtools/devtools-core/package.json:1]: All of the devtools changes can also be left as they were. Revert all under tools/devtools.

File: packages/tools/fluid-runner/package.json [hunk 3]
```diff
@@ -64,6 +70,7 @@
 		"check:biome": "biome check . --formatter-enabled=true",
 		"check:exports": "concurrently \"npm:check:exports:*\"",
 		"check:exports:bundle-release-tags": "api-extractor run --config api-extractor/api-extractor-lint-bundle.json",
+		"check:exports:cjs:index": "api-extractor run --config api-extractor/api-extractor-lint-index.cjs.json",
```
Comment (jason-ha) [inline packages/tools/fluid-runner/package.json:73]: Why is policy only requiring a cjs:index check? (question to self probably)

**Review by tylerbutler — APPROVED** [review]

**Pull request merged**